### PR TITLE
Fix evidence and unused arguments warnings except encoder/decoder

### PR DESCRIPTION
**Pull request by marcinwadon**

### CODE REVIEW

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala [hunk 1]
```diff
@@ -110,7 +109,7 @@ object BlockConsensusCell {
       _ <- cleanUpRoundData(ownProposal, ctx)
     } yield ()
 
-  private def broadcast[F[_]: Async, T <: Transaction: Encoder](
+  private def broadcast[F[_]: Async, T <: Transaction](
```
Comment (TheMMaciek) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala:112]: Where is that Encoder for `T <: Transaction` take from right now?

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/http/Routes.scala [hunk 2]
```diff
@@ -52,14 +50,14 @@ final case class Routes[F[_]: Async: KryoSerializer, T <: Transaction: Decoder:
...
-            case Left(errors) => BadRequest(ErrorResponse(errors.map(e => ErrorCause(e.show))).asJson)
+            case Left(errors) => BadRequest(ErrorResponse(errors.map(e => ErrorCause(e.show))))
             case Right(hash)  => Ok(("hash" ->> hash.value) :: HNil)
           }
       } yield response
 
     case GET -> Root / "transactions" / HashVar(hash) =>
       transactionStorage.find(hash).flatMap {
-        case Some(tx) => Ok(TransactionView(tx.signed.value, tx.hash, TransactionStatus.Waiting).asJson)
+        case Some(tx) => Ok(TransactionView(tx.signed.value, tx.hash, TransactionStatus.Waiting))
```
Comment (TheMMaciek) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/http/Routes.scala:60]: Transformation to json happens implicitly either way right?

Comment (marcinwadon) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/http/Routes.scala:60]: Exactly

Comment (marcinwadon) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/http/Routes.scala:60]: This is how it works in all other places

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala (hunk 1 above)
Comment (marcinwadon) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala:112]: hmm, good question. I see that the `PeerBlockConsensusInput` requires implicit encoder/decoder

Comment (marcinwadon) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala:112]: but looks like the tests are passing, so consensus and transactions are working correctly 🤔

Comment (kpudlik) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala:112]: For `DAGTransaction` it is derevo's `Encoder[DAGTransaction]`
image

For `CurrencyTransaction` as well:
image

Comment (TheMMaciek) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala:112]: Ok the `BlockConsensusClient` has the `Encoder` for that given transaction type injected during creation.

**Review by TheMMaciek — DISMISSED** [review]

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala (hunk 1 above)
Comment (kpudlik) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala:112]: Yeah, the decoder is injected at the very top in the `Main` method. There are no intermediate instances.

**Review by kpudlik — APPROVED** [review]

**Review by kpudlik — APPROVED** [review]

**Review by kpudlik — DISMISSED** [review]

**Review by kpudlik — APPROVED** [review]

**Pull request merged**